libraries/lib-music-information-retrieval/StftFrameProvider.cpp

```diff
@@ -35,8 +35,10 @@ double GetHopSize(int sampleRate, long long numSamples)
    // of two. This will spare us the need for resampling when we need to get the
    // autocorrelation of the ODF using an FFT.
    const auto idealHopSize = 0.01 * sampleRate;
-   const auto numFrames =
-      1 << (int)std::round(std::log2(numSamples / idealHopSize));
+   const int exponent = std::round(std::log2(numSamples / idealHopSize));
```

Comment:
Are you trying to handle corner cases?  You deal with zero for `numSamples`.

log2 returns -infinity for a zero argument, NaN for a negative, and idealHopSize might be zero.  I'm just trying to think of abusive inputs... the numbers come from virtual functions of `audio` below which have no postconditions.